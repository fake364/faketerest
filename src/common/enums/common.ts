export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
