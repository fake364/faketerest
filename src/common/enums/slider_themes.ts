export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
