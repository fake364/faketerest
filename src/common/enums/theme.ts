export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}