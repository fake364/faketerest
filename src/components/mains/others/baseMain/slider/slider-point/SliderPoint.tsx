import React from 'react';
import { SLIDER_THEMES } from '../../../../../../common/enums/slider_themes';
import { CommonTypes } from '../../../../../../common/types/common';
import clsx from 'clsx';

interface Props extends CommonTypes.ShownType {
	pointType: SLIDER_THEMES;
}

const SliderPoint: React.FC<Props> = ({ pointType, shownType }) => {
	const isCurrentPoint = pointType === shownType;
	const className = isCurrentPoint ? 'slider-active' : 'slider-not-active';
	return (
		<div slider-theme={pointType} className={clsx('last:m-0', className)} />
	);
};

export default SliderPoint;
