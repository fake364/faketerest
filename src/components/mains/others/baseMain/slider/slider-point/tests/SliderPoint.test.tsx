import React from 'react';
import { shallow } from 'enzyme';
import SliderPoint from '../SliderPoint';
import { SLIDER_THEMES } from '../../../../../../../common/enums/slider_themes';

describe('SliderPoint', () => {
  it('should display not active point if selected point doesnt match current', () => {
    const wrapper = shallow(
      <SliderPoint
        pointType={SLIDER_THEMES.NEW_OUTFIT}
        shownType={SLIDER_THEMES.HOME_DECOR}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should display active point if it is selected point', () => {
    const wrapper = shallow(
      <SliderPoint
        pointType={SLIDER_THEMES.HOME_DECOR}
        shownType={SLIDER_THEMES.HOME_DECOR}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
