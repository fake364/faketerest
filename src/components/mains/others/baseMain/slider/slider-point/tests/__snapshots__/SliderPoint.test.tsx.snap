// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderPoint should display active point if it is selected point 1`] = `
<div
  className="last:m-0 slider-active"
  slider-theme="homeDecor"
/>
`;

exports[`SliderPoint should display not active point if selected point doesnt match current 1`] = `
<div
  className="last:m-0 slider-not-active"
  slider-theme="newOutfit"
/>
`;
